core/src/main/scala/io/github/ghostbuster91/sttp/client3/ModelGenerator.scala

```diff
@@ -32,10 +38,24 @@ object ModelGenerator {
       coproduct: Coproduct
   ): Defn.Trait = {
     val defParams =
-      (coproduct.properties ++ coproduct.additionalProperties).map(_.asDef)
+      (coproduct.properties ++ coproduct.additionalProperties)
+        .filterNot { property =>
+          isDiscriminatorProperty(property, List(coproduct))
+        }
+        .map(_.asDef)
     q"""sealed trait ${coproduct.name.typeName} {
             ..$defParams
         }
         """
   }
+
+  private def isDiscriminatorProperty(
+      property: ParameterRef,
+      coproducts: List[Coproduct]
+  ): Boolean =
+    coproducts
+      .flatMap(_.discriminator)
+      .exists { discriminator =>
+        property.paramName.term.value == discriminator.fieldName
+      }
```

Comment:
Every property that is a discriminator in any of the Products parents is marked as a discriminator property and thus not created as part of the model.

There could be an edge case in which a Product is a child of multiple Coproducts that define different param names for their discriminators, in which case you might want to have all these params as part of the model.

I see this as an anti-pattern that should be avoided.